### PR TITLE
WIP: Elm common lib

### PR DESCRIPTION
**Pull request by La0**
* TaskclusterLogin
* BugzillaLogin
* Hawk (in progress)

Usable from shipit_frontend AppTest

### CODE REVIEW

File: lib/elm_common/TaskclusterLogin.elm [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    , accessToken = fromJust (Dict.get "accessToken" query)
+    , certificate =
+             case Dict.get "certificate" query of
+                 Just certificate ->
+                     Result.toMaybe <| decodeCertificate certificate
+                 Nothing -> Nothing
+    }
+
+-- Views
+view model =
```
Comment (garbas) [inline lib/elm_common/TaskclusterLogin.elm]: i think we should remove this and this should be implemented in a Parent component

File: lib/elm_common/TaskclusterLogin.elm [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+  case model.credentials of
+    Just user ->
+      div [] [text ("Logged in as " ++ user.clientId)]
+    Nothing ->
+      div [] (viewLogin)
+
+viewLogin =
```
Comment (garbas) [inline lib/elm_common/TaskclusterLogin.elm]: viewLogin should not provide any html structure but only attribute which triggers ``Login loginUrl`` message.

File: lib/elm_common/BugzillaLogin.elm [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+    viewEditor user,
+    hr [] [],
+    div [class "alert alert-info"] [
+      strong [] [text "Heads up !"],
+      p [] [text "Your Bugzilla credentials are stored only in your browser, using localStorage. They are NOT sent to any backend for storage."]
+    ]
+  ]
+
+viewEditor: Model -> Html Msg
+viewEditor user =
```
Comment (garbas) [inline lib/elm_common/BugzillaLogin.elm]: ``BugzillaLogin.elm`` should not enforce any html structure. more reusable would be to provide those tree messages as separate functions which return Html attribute

- ``onSubmit SaveCredentials``
- ``onInput UpdateLogin``
- ``onInput UpdateToken``

**Comment by La0** [comment]
Closing to use common branch

**Pull request closed without merging**